Assets/Scripts/GameLogic/QuestionGenerator.cs

```diff
@@ -21,13 +20,67 @@ void Start()
         a1 = GameObject.Find(Constants.ANSWER_ONE_OVERLAY).GetComponent<TMP_Text>();
         a2 = GameObject.Find(Constants.ANSWER_TWO_OVERLAY).GetComponent<TMP_Text>();
         a3 = GameObject.Find(Constants.ANSWER_THREE_OVERLAY).GetComponent<TMP_Text>();
-
-        // Generate answer
-        minNum = 0;
-        maxNum = 10;
-        X = Random.Range(minNum,maxNum);
-        Y = Random.Range(minNum,maxNum);
-        answer = X + Y;
+        
+        // Generate question
+        int level = LevelController.level;
+        int chooseDifficulty = Random.Range(0, 1 + level/5);
+        string op = "";
+        switch (chooseDifficulty) {
+            case 0: // single digit addition
+                X = Random.Range(0, 9);
+                Y = Random.Range(0, 9);
+                answer = X + Y;
+                op = Constants.ADDITION;
+                break;
+            case 1: // double/single digit addition
+                X = Random.Range(0, 99);
+                Y = Random.Range(0, 99);
+                answer = X + Y;
+                op = Constants.ADDITION;
+                break;
+            case 2: // single digit subtraction
+                X = Random.Range(0, 9);
+                Y = Random.Range(0, 9);
+                answer = X - Y;
+                op = Constants.SUBTRACTION;
+                break;
+            case 3: // double/single digit subtraction
+                X = Random.Range(0, 99);
+                Y = Random.Range(0, 99);
+                answer = X - Y;
+                op = Constants.SUBTRACTION;
+                break;
+            case 4: // single digit multiplication
+                X = Random.Range(0, 9);
+                Y = Random.Range(0, 9);
+                answer = X * Y;
+                op = Constants.MULTIPLICATION;
+                break;
+            case 5: // double/single digit multiplication
+                X = Random.Range(1, 99);
+                Y = Random.Range(0, 1000/X); // keep answers below 1000 to maintain simplicity
+                answer = X * Y;
+                op = Constants.MULTIPLICATION;
+                break;
+            case 6: // single digit division
+                answer = Random.Range(0, 9);
+                Y = Random.Range(1, 9);
+                X = answer * Y;
+                op = Constants.DIVISION;
+                break;
+            case 7: // double/single digit division
+                answer = Random.Range(0, 99);
+                Y = Random.Range(1, 1000/99); // keep answers below 1000 to maintain simplicity
+                X = answer * Y;
+                op = Constants.DIVISION;
+                break;
+            default:
```

Comment:
After reaching level 45+ a lot of the questions were just 0 + 0 because of the default case and the range for chooseDifficulty exceeding case 7, this should change so that it selects one of the more challenging questions by default.